proxy/api/src/main.rs

```diff
@@ -83,15 +83,57 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let bin_dir = config::bin_dir()?;
     coco::git_helper::setup(&proxy_path, &bin_dir).expect("Git remote helper setup failed");
 
-    log::info!("Starting API");
     let ctx = context::Ctx::from(context::Context {
         peer_api,
         keystore,
         store,
     });
+
+    {
+        let ctx = ctx.clone();
+        log::info!("Starting Announcement watcher");
+        tokio::task::spawn(announcement_watcher(ctx));
+    }
+
+    log::info!("Starting API");
     let api = http::api(ctx, args.test);
 
     warp::serve(api).run(([127, 0, 0, 1], 8080)).await;
 
     Ok(())
 }
+
+async fn announcement_watcher(ctx: context::Ctx) {
+    // TODO(xla): Take interval timings from config/settings.
+    let mut timer = tokio::time::interval(std::time::Duration::from_secs(10));
+
+    loop {
+        timer.tick().await;
+
+        if let Err(err) = announce(ctx.clone()).await {
+            log::info!("Announcement watcher errored: {:?}", err);
```

Comment:
`log::error` or `log::warn` would be better no?

Reply:
Will be addressed as part of #854.

Reply:
That ticket doesn't mention auditing the levels of the logging messages

Reply:
> The proxy codebase should be overhauled and be equipped with a sound logging strategy

Reply:
A bit vague but I think I get where you're coming from